jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -581,6 +586,52 @@ def set_model_component(self, model_component_label, parameter, value=None, fixe
 
         return parameter
 
+    def vue_reestimate_model_parameters(self, model_component_label=None, **kwargs):
+        self.reestimate_model_parameters(model_component_label=model_component_label)
+
+    def reestimate_model_parameters(self, model_component_label=None):
+        """
+        Re-estimate all free parameters in a given model component given the currently selected
+        data and subset selections.
+
+        Parameters
+        ----------
+        model_component_label : str or None.
+            The label given to the existing model component.  If None, will iterate over all model
+            components.
+        """
+        if model_component_label is None:
+            ret = []
+            for model_comp in self.component_models:
+                ret.append(self.reestimate_model_parameters(model_comp["id"]))
+            return ret
```

Comment:
Any reason this can't be shortened?

```suggestion
            return [self.reestimate_model_parameters(model_comp["id"]) for model_comp in self.component_models]
```

Reply:
good catch - I think I had more logic in the for-loop originally, but definitely should simplify to this.